homeassistant/components/verisure/lock.py

```diff
@@ -128,13 +128,19 @@ def extra_state_attributes(self) -> dict[str, str]:
 
     async def async_unlock(self, **kwargs: Any) -> None:
         """Send unlock command."""
-        code = kwargs.get(ATTR_CODE)
+        code: str = kwargs.get(
+            ATTR_CODE,
+            self._lock_option_default_code,
```

Comment:
@gjohansson-ST is this how you intended using the default code?

Reply:
No, as it should be covered here https://github.com/home-assistant/core/blob/dev/homeassistant/components/lock/__init__.py#L102-L111
That is then passed to the integration `async_unlock()` call which should then be feed with the code (either input or the default configured code).

Since you have this lock you can perhaps add a bit of debugging and see if it gets lost or something strange on the passing through?
I can't see anything in the code that would make the code disappear if there is one.

Reply:
@gjohansson-ST The default code **not** passed in the code you linked to, it is only verified that it matches the code format _if_ no code is given in the service call. I am happy to fix this in the Lock component instead, but I wanted to check if this was the intended way or not.

Above code is similar to how this is done in the Matter integration: https://github.com/home-assistant/core/blob/dev/homeassistant/components/matter/lock.py#L92-L95

If the Lock component is changed, the Matter integration should be simplified.

Reply:
OMG. How did I miss that!!! 😱 
Yes, we should swing this PR to fix the problem in `lock` instead. The integrations should not have to care about this themselves.
There must be a clumsy bug in the tests too then...
Can fix the `matter` integration (and others?) in a follow-up PR.

Reply:
PR swinged. I only found the workaround in the `matter` integration, but once this PR is merged I can do another search.